### ISSUE TITLE
Metrics - Cloudwatch - Histogram

### ISSUE THREAD

**Issue by nelz9999** [go-kit/kit#660]
Heya folks!

I'm using the metrics package with CloudWatch, and I'd like to propose an addition.

The background:
I can see some use for how the current implementation uses in-memory histogram to emit stuff to CloudWatch, but really I don't need that... I can get the percentiles I need via native CloudWatch. (https://aws.amazon.com/about-aws/whats-new/2016/11/amazon-cloudwatch-adds-support-for-percentiles/)

So, in the short-term, I've just been emitting my latency-related metrics as `Counter`s, but that feels like I'm fighting the expressiveness added by `go-kit`. (I.e. "I expect this metric to be of a latency/duration type.")

What would you all feel about me adding something like a `NewNativeHistogram(...)` that conforms to the `Histogram` API, but actually just puts measurements into the `Counter` buckets?

Thanks!

**Comment by peterbourgon** [comment]
Happy to take such a PR! Please make sure to include some basic test coverage.

**Comment by nelz9999** [comment]
Urgh... This is not working. 😞 

On my first attempt, the initial implementation seemed to work okay... But then I got blocked trying to write the unit test. (The test harnesses don't really have a way of "crossing the streams" between counter-like behavior and histogram-like behavior.)

My second attempt, I tried to just create a converter on the `cloudwatch.Counter` type, but I realized I couldn't make the `With(...)` function work, unless I changed the signature to return the concrete struct rather than the interface.

At this point I'm thinking that I'll just make my own Counter->Histogram wrapper, so that I don't expose y'all to my weird corner case.
